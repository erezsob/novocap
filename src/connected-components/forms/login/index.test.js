import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import LoginForm from './';

const mockStore = configureStore([thunk]);
const initialState = {};
const store = mockStore(initialState);

describe('<LoginForm />', () => {
  it('renders', () => {
    const wrapper = shallow(<LoginForm store={store} />);
    expect(wrapper).toHaveLength(1);
  });

  it('renders properly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <LoginForm />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
