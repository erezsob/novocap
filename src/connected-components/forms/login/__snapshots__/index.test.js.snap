// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> renders properly 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <div
    className="container"
  >
    <label
      className="label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-1"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </label>
    <div
      className="inputContainer"
    >
      <input
        className="input"
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Username"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="container"
  >
    <label
      className="label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-1"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12.65 10C11.83 7.67 9.61 6 7 6c-3.31 0-6 2.69-6 6s2.69 6 6 6c2.61 0 4.83-1.67 5.65-4H17v4h4v-4h2v-4H12.65zM7 14c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
        />
      </svg>
    </label>
    <div
      className="inputContainer"
    >
      <input
        className="input"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
  </div>
  <button
    className="button bigger"
    onClick={undefined}
  >
    <span
      className="text"
    >
      Login
    </span>
  </button>
</form>
`;
